src/fediblockhole/__init__.py

```diff
@@ -721,6 +721,40 @@ def save_domain_block_audit_to_file(
         for key, value in sorted_list:
             writer.writerow(value)
 
+def resolve_replacements(endpoints: list[dict]) -> list[dict]:
+    """Resolve any replacement tokens in the list of endpoints"""
+    resolved = []
+    for item in endpoints:
+        item = dict(**item)
+        if 'token' in item and 'token_env_var' in item:
+            log.warning(f"Both `token` and `token_env_var` have been provided; using the explicit token for {item.get("domain", "the entry")}")
+            # we take the token that's explicitly stated, even if there's also an environment
+            # variable set. Delete the token_env_var key
+            del item['token_env_var']
+
+        elif 'token' in item:
+            ...
+
+        elif 'token_env_var' in item:
+            value = os.getenv(item['token_env_var'])
+            if value is None:
+                raise ValueError(f"Environment variable '{item["token_env_var"]}' not set.")
+
+            item['token'] = value
+
+        else:
+            # lastly, try look for a default token.
```

Comment:
Grammar tweak to comment would be useful. Not a blocker for merge.

Reply:
I admit, I don't see the grammar issue you're referring to; feel free to use the "suggest" button to propose the change you want?

Reply:
`# lastly, try to look for a default token`

It was mostly a note for myself to follow up post-merge rather than block progress.